ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -58,56 +59,108 @@ func NewReadWriter() (*YAMLReadWriter, error) {
+	cliConfig.Cluster = cluster.Cluster
+	cliConfig.ComposeServiceNamePrefix = cluster.ComposeServiceNamePrefix
+	return nil
+
+}
+
+// readClusterConfig does all the work to read and parse the yaml cluster config
+func readProfileConfig(path string, profileConfigKey string, cliConfig *CLIConfig) error {
+	// read cluster file
+	dat, err := ioutil.ReadFile(path)
 	if err != nil {
-		return errors.Wrapf(err, "Error reading yaml file %s", yamlPath)
+		return errors.Wrap(err, "Failed to read config file: "+path)
 	}
-	if err = yaml.Unmarshal(dat, cliConfig); err != nil {
-		return errors.Wrapf(err, "Error parsing yaml file %s", yamlPath)
+
+	config := ProfileConfig{Profiles: make(map[string]Profile)}
+	if err = yaml.Unmarshal(dat, &config); err != nil {
+		return errors.Wrap(err, "Failed to parse yaml file: "+path)
 	}
 
-	// convert yaml to a map (replaces IsKeyPresent functionality)
-	if err = yaml.Unmarshal(dat, &configMap); err != nil {
-		return errors.Wrapf(err, "Error parsing yaml file %s", yamlPath)
+	// get the correct cluster
+	var chosenProfile string
+	if profileConfigKey == "" {
+		chosenProfile = config.Default
+	} else {
+		chosenProfile = profileConfigKey
 	}
+	profile := config.Profiles[chosenProfile]
+
+	// Get the info out of the cluster
+	cliConfig.AWSSecretKey = profile.AWSSecretKey
+	cliConfig.AWSAccessKey = profile.AWSAccessKey
 
 	return nil
+
 }
 
-// GetConfig gets the ecs-cli config object from the config file.
+// GetConfigs gets the ecs-cli config object from the config file(s).
 // map contains the keys that are present in the config file (maps string field name to string field value)
 // map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
 // cause certain keys to be mapped to maps of keys
-func (rdwr *YAMLReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
-	to := new(CLIConfig)
-	configMap := make(map[interface{}]interface{})
-	// read the raw bytes of the config file
-	path := configPath(rdwr.destination)
-
-	if errYAML := readYAML(path, configMap, to); errYAML == nil {
-		// File is YAML
-		return to, configMap, nil
-	} else if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
+// This function either reads the old single configuration file
+// Or if the new files are present, it reads from them instead
+func (rdwr *YAMLReadWriter) GetConfigs(clusterConfig string, profileConfig string) (*CLIConfig, error) {
+	cliConfig := &CLIConfig{} // read the raw bytes of the config file
+	profilePath := credentialsFilePath(rdwr.destination)
+	configPath := configFilePath(rdwr.destination)
+
+	// Try to read the config as YAML
+	// throw error if it fails
+	errYAMLConfig := readClusterConfig(configPath, clusterConfig, cliConfig)
+
+	if _, err := os.Stat(profilePath); err == nil {
+		// credentials file exists- so that means we are using the new style configs
+		err := readProfileConfig(profilePath, profileConfig, cliConfig)
```

Comment:
Then this works fine. If the profile file does not exist then no error is thrown. There is no issue here.